lib/src/utilities/extract-nested-types.ts

```diff
@@ -0,0 +1,143 @@
+import assertNever from "assert-never";
+import { flatten } from "lodash";
+import { TypeNode } from "../models/nodes";
+import { ObjectType, TypeKind, UnionType } from "../models/types";
+import { resolveType } from "../verifiers/utilities/type-resolver";
+
+export function maybeResolveRef(
```

Comment:
Could we delete this function and use https://github.com/airtasker/spot/blob/master/lib/src/verifiers/utilities/type-resolver.ts#L41 instead?